### DISCUSSION TITLE
before all dag run, to run a hook

### DISCUSSION THREAD

**Discussion by chuxiangfeng** [apache/airflow#27707] [Q&A] [answered]
i want get to do something with info (dag_id, task_ids for dag) before all dags run each run,how can i do ? is any hook can i use ?

**Comment by potiuk** [comment, marked as answer]
Look at https://airflow.apache.org/docs/apache-airflow/stable/concepts/cluster-policies.html - this is all there is.

**Reply by chuxiangfeng** [reply]
> Look at https://airflow.apache.org/docs/apache-airflow/stable/concepts/cluster-policies.html - this is all there is.

It does what I want it to do, but unfortunately it doesn't fire every time the dag runs or the dag structure changes, and seems to be executed only on the first launch
